internal/datastore/postgres/watch.go

```diff
@@ -197,9 +219,18 @@ func (pgd *pgDatastore) Watch(
 	return updates, errs
 }
 
-func (pgd *pgDatastore) getNewRevisions(ctx context.Context, afterTX postgresRevision) ([]postgresRevision, error) {
+func (pgd *pgDatastore) getNewRevisions(ctx context.Context, afterTX postgresRevision, returnHeadRevision bool) ([]postgresRevision, *postgresRevision, error) {
```

Comment:
personally I think I'd find it easier to come back to this code if it were split into `getNewRevisions` and `getNewRevisionsWithHead`

Then you wouldn't have to do any nil checking for the `optionalHeadRevision` (because you'd just return an error from `getNewRevisionsWithHead`)

Reply:
so this method is only used in 1 place, that's the postgres watch loop. And we pass  `returnHeadRevision` arg to it as true if the client requested `datastore.WatchCheckpoints`, which is optional (e.g. SpiceDB Watch API does not currently return checkpoints), so it was added as an optimization in case someone does not need checkpoints.

so that refactor would be useful if we used the method in 2 different places, and one of them needed either always head, or never head, but that's not the case - it's entirely driven by the `Datastore.Watch` call inputs. Does that make sense?